### ISSUE TITLE
[BUG] CUDA boxing: resolve_conj clears the conjugate bit without materializing the conjugate

### ISSUE THREAD

**Issue by FlagQuantum** [flagos-ai/Torch-FL#209]
## Bug Description

On the CUDA-backed `flagos:0` route, PyTorch's lazy conjugation semantics
appear to be incorrect.

`torch.conj(x)` sets the conjugate bit, but
`torch.resolve_conj(torch.conj(x))` clears the bit without negating the
imaginary component.

`torch.conj_physical(x)` produces the correct result, and the same operations
all work correctly on native CUDA.

## Environment

- **Platform**: CUDA, NVIDIA A800-SXM4-80GB
- **Python Version**: 3.12.13
- **PyTorch Version**: 2.11.0+cu130
- **torch_fl Version/Branch**:
  - Package version: `0.1.0`
  - Source revision: `1b19383c24d9aa54637dd23557a5a20ef6d52ec5`
  - Git describe: `v0.1.0-52-g1b19383`
- **CUDA/MACA/CANN Version**:
  - PyTorch CUDA runtime: `13.0`
  - Container CUDA version: `13.0.2`
- **FlagGems Version**: Not applicable; both FlagGems kernel paths were disabled

The issue was reproduced on the exact source revision above. It has not yet
been retested on the latest Torch-FL main branch.

## Build Configuration

```bash
ACCELERATOR=cuda \
CUDA_KERNEL=1 \
FLAGGEMS_KERNEL=OFF \
FLAGGEMS_PYTHON=OFF \
python -m pip install \
  --no-build-isolation \
  --no-deps \
  -e .
```

## Runtime Configuration

```bash
export CUDA_VISIBLE_DEVICES=0
export FLAGGEMS_KERNEL=OFF
export FLAGGEMS_PYTHON=OFF
```

Torch-FL automatically reported loading:

```text
torch_fl/configs/backends_cuda.conf
```

## Steps to Reproduce

```python
import torch
import torch_fl


source_values = [
    -0.4 - 0.6j,
    0.25 + 0.5j,
]

expected = torch.tensor(
    [
        -0.4 + 0.6j,
        0.25 - 0.5j,
    ],
    dtype=torch.complex128,
)


def check(device):
    x = torch.tensor(
        source_values,
        dtype=torch.complex128,
        device=device,
    )

    lazy = torch.conj(x)
    resolved = torch.resolve_conj(lazy)
    physical = torch.conj_physical(x)

    lazy_cpu = lazy.cpu()
    resolved_cpu = resolved.cpu()
    physical_cpu = physical.cpu()

    print(f"\ndevice={device}")
    print("source:", x.cpu())
    print("conj:", lazy_cpu)
    print("conj.is_conj:", lazy.is_conj())
    print("resolve_conj:", resolved_cpu)
    print("resolve_conj.is_conj:", resolved.is_conj())
    print("conj_physical:", physical_cpu)

    print(
        "conj error:",
        torch.max(torch.abs(lazy_cpu - expected)).item(),
    )
    print(
        "resolve_conj error:",
        torch.max(torch.abs(resolved_cpu - expected)).item(),
    )
    print(
        "conj_physical error:",
        torch.max(torch.abs(physical_cpu - expected)).item(),
    )


check("cuda:0")

torch.flagos.set_device(0)
check("flagos:0")
```

## Expected Behavior

For the input:

```text
[-0.4-0.6j, 0.25+0.5j]
```

all three conjugation paths should produce:

```text
[-0.4+0.6j, 0.25-0.5j]
```

In particular:

```python
lazy = torch.conj(x)
resolved = torch.resolve_conj(lazy)
physical = torch.conj_physical(x)

assert lazy.is_conj()
assert not resolved.is_conj()
assert torch.allclose(lazy.cpu(), expected)
assert torch.allclose(resolved.cpu(), expected)
assert torch.allclose(physical.cpu(), expected)
```

`torch.resolve_conj(torch.conj(x))` should be numerically equivalent to
`torch.conj_physical(x)`.

## Actual Behavior

Native CUDA produces the expected results:

```text
device=cuda:0

source:
tensor([-0.4000-0.6000j, 0.2500+0.5000j], dtype=torch.complex128)

conj:
tensor([-0.4000+0.6000j, 0.2500-0.5000j], dtype=torch.complex128)

conj.is_conj:
True

resolve_conj:
tensor([-0.4000+0.6000j, 0.2500-0.5000j], dtype=torch.complex128)

resolve_conj.is_conj:
False

conj_physical:
tensor([-0.4000+0.6000j, 0.2500-0.5000j], dtype=torch.complex128)

conj error: 0.0
resolve_conj error: 0.0
conj_physical error: 0.0
```

The `flagos:0` route produces:

```text
device=flagos:0

source:
tensor([-0.4000-0.6000j, 0.2500+0.5000j], dtype=torch.complex128)

conj:
tensor([-0.4000-0.6000j, 0.2500+0.5000j], dtype=torch.complex128)

conj.is_conj:
True

resolve_conj:
tensor([-0.4000-0.6000j, 0.2500+0.5000j], dtype=torch.complex128)

resolve_conj.is_conj:
False

conj_physical:
tensor([-0.4000+0.6000j, 0.2500-0.5000j], dtype=torch.complex128)

conj error: 1.2
resolve_conj error: 1.2
conj_physical error: 0.0
```

Summary:

| Route | `torch.conj` | `torch.resolve_conj` | `torch.conj_physical` |
|---|---:|---:|---:|
| Native CUDA | correct, error `0` | correct, error `0` | correct, error `0` |
| `flagos:0` | incorrect, error `1.2` | incorrect, error `1.2` | correct, error `0` |

No exception or traceback is raised. This is a silent numerical correctness
failure.

## Additional Context

The results indicate that:

1. `torch.conj` sets the conjugate bit on `flagos:0`.
2. `torch.resolve_conj` clears the bit.
3. Clearing the bit does not physically negate the imaginary component.
4. `torch.conj_physical` works correctly.
5. Native CUDA works correctly on the same A800.
6. The problem reproduces on a single device without distributed communication.
7. Downstream differential tests observed the same behavior with both
   `complex64` and `complex128`.

This suggests that the physical conjugation kernel is functional and that the
problem is in the lazy-conjugation resolution path.

One possible area to inspect is the handling of the Conjugate dispatch key and
view/alias semantics across the PrivateUse1-to-CUDA boxing boundary. This is
only an initial hypothesis; the precise point where the conjugate state is lost
has not yet been instrumented.

The issue affects expressions such as:

```python
torch.real(torch.sum(torch.conj(x) * y))
```

Such expressions may silently produce incorrect values and gradients while
retaining apparently valid dtype and device metadata.

## Checklist

- [ ] I have searched existing issues to avoid duplicates
- [x] I have provided complete environment information
- [x] I have included a minimal reproducible example
- [x] I have included the full error traceback
  - Not applicable: this is a silent numerical correctness failure and no
    exception or traceback is produced.
